pkg/controller/version/docker.go

```diff
@@ -24,62 +27,85 @@ func NewDockerVersionChecker(currentImage, currentImageId string, dockerConfig *
-	reference, err := name.ParseReference(strings.TrimPrefix(dockerVersionChecker.currentImageId, "docker-pullable://"))
+	latestReference, err := alltransports.ParseImageName("docker://" + dockerVersionChecker.currentImage)
 	if err != nil {
 		return false, err
 	}
 
-	latestReference, err := name.ParseReference(latestImageName)
+	latestDigest, err := dockerVersionChecker.getDigest(latestReference)
 	if err != nil {
 		return false, err
 	}
 
-	registryURL := "https://" + reference.Context().RegistryStr()
-	authOption := getAuthOption(dockerVersionChecker.dockerConfig, registryURL)
-
-	latestDigest, err := getDigest(latestReference, authOption)
+	//Using ImageID instead of Image because ImageID contains digest of image that is used while Image only contains tag
+	currentReference, err := alltransports.ParseImageName(transportImageName)
+	//reference, err := name.ParseReference(strings.TrimPrefix(dockerVersionChecker.currentImageId, "docker-pullable://"))
 	if err != nil {
 		return false, err
 	}
 
-	currentDigest, err := getDigest(reference, authOption)
+	currentDigest, err := dockerVersionChecker.getDigest(currentReference)
 	if err != nil {
 		return false, err
 	}
 
 	return currentDigest == latestDigest, nil
 }
 
-func getDigest(reference name.Reference, authOption remote.Option) (string, error) {
-	img, err := remote.Image(reference, authOption)
+func (dockerVersionChecker *DockerVersionChecker) getDigest(ref types.ImageReference) (digest.Digest, error) {
+	systemContext := dockerVersionChecker.makeSystemContext(ref.DockerReference())
+	imageSource, err := ref.NewImageSource(context.TODO(), systemContext)
+	if err != nil {
+		return "", err
+	}
+	defer closeImageSource(imageSource)
+
+	imageManifest, _, err := imageSource.GetManifest(context.TODO(), nil)
 	if err != nil {
 		return "", err
 	}
 
-	digest, err := img.Digest()
+	imageDigest, err := manifest.Digest(imageManifest)
 	if err != nil {
 		return "", err
 	}
 
-	return digest.Hex, nil
+	return imageDigest, nil
+}
+
+func closeImageSource(source types.ImageSource) {
+	if source != nil {
+		// Swallow error
+		_ = source.Close()
+	}
 }
 
-func getAuthOption(dockerConfig *parser.DockerConfig, registryURL string) remote.Option {
-	if dockerConfig == nil {
-		return remote.WithAuthFromKeychain(authn.DefaultKeychain)
+func (dockerVersionChecker *DockerVersionChecker) makeSystemContext(dockerReference reference.Named) *types.SystemContext {
+	if dockerReference == nil || dockerVersionChecker.dockerConfig == nil {
+		return &types.SystemContext{}
 	}
 
-	credentials, hasCredentials := dockerConfig.Auths[registryURL]
+	registryName := strings.Split(dockerReference.Name(), "/")[0]
+	credentials, hasCredentials := dockerVersionChecker.dockerConfig.Auths[registryName]
+
 	if !hasCredentials {
-		return remote.WithAuthFromKeychain(authn.DefaultKeychain)
+		registryURL := "https://" + registryName
```

Comment:
To reiterate my previous comment, I don't have the protocol `https://` on the Dockerconfig from my local machine, are you certain you'll see it on Auths's map?

Reply:
I have added a check so both variants are covered:
```
registryName := strings.Split(dockerReference.Name(), "/")[0]
credentials, hasCredentials := dockerVersionChecker.dockerConfig.Auths[registryName]
 ```
checks without protocol. If it does not find it 
```
registryURL := "https://" + registryName
credentials, hasCredentials = dockerVersionChecker.dockerConfig.Auths[registryURL]
```
checks with prepended protocol

Reply:
Ah, missed that, sorry. All good then.